pkg/zson/zson.go

```diff
@@ -48,3 +49,33 @@ type Batch interface {
 	//XXX span should go in here?
 	Span() nano.Span
 }
+
+func CopyWithContext(ctx context.Context, dst WriteCloser, src Reader) error {
+	var err error
+	for ctx.Err() == nil {
+		var rec *Record
+		rec, err = src.Read()
+		if err != nil || rec == nil {
+			break
+		}
+		err = dst.Write(rec)
+		if err != nil {
+			break
+		}
+	}
+	dstErr := dst.Close()
+	switch {
+	case err != nil:
+		return err
+	case dstErr != nil:
+		return dstErr
+	default:
+		return ctx.Err()
+	}
+}
+
+// Copy copies src to dst a la io.Copy.  The src reader is read from
+// while the dst writer is written to and closed.
+func Copy(dst WriteCloser, src Reader) error {
```

Comment:
Do you have an actual use case for this? Would it be better to just copy the raw zval from one buffer to another?

Reply:
I use it in the tests I added, but it could be generally useful.  The reader can be parsing one format and the writer can be writing another.  The zval is being copied from the reader to the writer.

Reply:
Also, zq could do this for "*" with no processing instead of building a flowgraph.